### PR TITLE
Db

### PR DESCRIPTION
**Pull request by dartuso**
Implemented Entities from Relational Model in android room database format.
Still need to do some foreign keys, and review primary key tags.

### CODE REVIEW

File: app/src/main/java/com/group15/fitnesstracker/db/BodyMeasure_Recording.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.group15.fitnesstracker.db;
+
+import java.util.Date;
+
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class BodyMeasure_Recording {
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/BodyMeasure_Recording.java]:
```suggestion
public class BodyMeasureRecording {
```

File: app/src/main/java/com/group15/fitnesstracker/db/BodyMeasure_Recording.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @PrimaryKey
+    public int recording_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/BodyMeasure_Recording.java]:
```suggestion
    public int recordingId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Goal.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Goal {
+    @PrimaryKey
+    @ColumnInfo(name = "id")
+    public int id;
+
+    public String goal_description;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Goal.java]:
```suggestion
    public String goalDescription;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Goal.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    public Date completion_date;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Goal.java]:
```suggestion
    public Date completionDate;
```

File: app/src/main/java/com/group15/fitnesstracker/db/History.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class History {
+    @PrimaryKey
+    //TODO: @ForeignKey()
+    public int user_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/History.java]:
```suggestion
    public int userId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/History.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    @PrimaryKey
+    //TODO: @ForeignKey()
+    public int workout_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/History.java]:
```suggestion
    public int workoutId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Micronutrient_Recording.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.PrimaryKey;
+
+
+@Entity(foreignKeys = @ForeignKey(entity = Nutrition_Recording.class,
+                                  parentColumns = "recording_id",
+                                  childColumns = " recording_id"))
+public class Micronutrient_Recording {
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Micronutrient_Recording.java]:
```suggestion
public class MicronutrientRecording {
```

File: app/src/main/java/com/group15/fitnesstracker/db/Micronutrient_Recording.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    @PrimaryKey
+    public int recording_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Micronutrient_Recording.java]:
```suggestion
    public int recordingId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Micronutrient_Recording.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+    public micronutrient[] micronutrients;
+}
+class micronutrient {
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Micronutrient_Recording.java]: ?? why lowercase
```suggestion
class Micronutrient {
```

File: app/src/main/java/com/group15/fitnesstracker/db/Nutrition_Recording.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
+package com.group15.fitnesstracker.db;
+
+import java.util.Date;
+
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Nutrition_Recording {
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Nutrition_Recording.java]:
```suggestion
public class NutritionRecording {
```

File: app/src/main/java/com/group15/fitnesstracker/db/Nutrition_Recording.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    @PrimaryKey
+    public int recording_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Nutrition_Recording.java]:
```suggestion
    public int recordingId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Nutrition_Recording.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+    public Date date;
+
+    public double calories;
+
+    public macronutrient macronutrient;
+}
+
+class macronutrient {
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Nutrition_Recording.java]: same here
```suggestion
class Macronutrient {
```

File: app/src/main/java/com/group15/fitnesstracker/db/Records.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Records {
+    @PrimaryKey
+//    @ForeignKey()
+    public String user_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Records.java]:
```suggestion
    public String userId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Records.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+    @PrimaryKey
+//    @ForeignKey()
+    public String record_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Records.java]:
```suggestion
    public String recordId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Schedule.java [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Schedule {
+    @PrimaryKey
+//    @ForeignKey()
+    public int workout_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Schedule.java]:
```suggestion
    public int workoutId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Schedule.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+    @PrimaryKey
+//    @ForeignKey()
+    public int trainer_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Schedule.java]:
```suggestion
    public int trainerId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Schedule.java [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+
+    @PrimaryKey
+//    @ForeignKey()
+    public int user_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Schedule.java]:
```suggestion
    public int userId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Schedule.java [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+
+    @PrimaryKey
+//    @ForeignKey()
+    public int recording_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Schedule.java]:
```suggestion
    public int recordingId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Set_Exercise.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
+package com.group15.fitnesstracker.db;
+
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Set_Exercise {
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Set_Exercise.java]:
```suggestion
public class SetExercise {
```

File: app/src/main/java/com/group15/fitnesstracker/db/Set_Exercise.java [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+    @PrimaryKey
+    public String name;
+
+//    @ForeignKey()
+    public int workout_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Set_Exercise.java]:
```suggestion
    public int workoutId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Timed_Exercise.java [hunk 21]
```diff
@@ -0,0 +1,18 @@
+package com.group15.fitnesstracker.db;
+
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Timed_Exercise {
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Timed_Exercise.java]:
```suggestion
public class TimedExercise {
```

File: app/src/main/java/com/group15/fitnesstracker/db/Timed_Exercise.java [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+    @PrimaryKey
+    public String name;
+
+//    @ForeignKey()
+    public int workout_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Timed_Exercise.java]:
```suggestion
    public int workoutId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Trainer.java [hunk 23]
```diff
@@ -0,0 +1,16 @@
+package com.group15.fitnesstracker.db;
+
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Trainer {
+    @PrimaryKey
+    public String trainer_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Trainer.java]:
```suggestion
    public String trainerId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Trains.java [hunk 24]
```diff
@@ -0,0 +1,12 @@
+package com.group15.fitnesstracker.db;
+
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+
+@Entity
+public class Trains {
+    //    @ForeignKey()
+    public int trainer_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Trains.java]:
```suggestion
    public int trainerId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Trains.java [hunk 25]
```diff
@@ -0,0 +1,12 @@
...
+//    @ForeignKey()
+    public int user_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Trains.java]:
```suggestion
    public int userId;
```

File: app/src/main/java/com/group15/fitnesstracker/db/Workout.java [hunk 26]
```diff
@@ -0,0 +1,15 @@
+package com.group15.fitnesstracker.db;
+
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+
+@Entity
+public class Workout {
+    @PrimaryKey
+    public int workout_id;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/Workout.java]:
```suggestion
    public int workoutId;
```

**Review by awave1 — CHANGES_REQUESTED** [review]
looks good, just gotta fix the formatting (need to use CamelCase)

**Review by awave1 — APPROVED** [review]

**Comment by dartuso** [comment]
Thank u mr project manager

File: app/src/main/java/com/group15/fitnesstracker/db/BodyMeasureRecording.java [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+
+    public double bodyfat;
+
+    public double weight;
+
+    public BodyPartMeasurement[] bodyPartMeasurements;
+}
+
+class BodyPartMeasurement {
+    public String Bodypart;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/BodyMeasureRecording.java]:
```suggestion
    public String bodypart;
```

File: app/src/main/java/com/group15/fitnesstracker/db/BodyMeasureRecording.java [hunk 28]
```diff
@@ -0,0 +1,25 @@
...
+    public double Bodypart_size;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/BodyMeasureRecording.java]:
```suggestion
    public double bodypartSize;
```

**Review by awave1 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/group15/fitnesstracker/db/User.java [hunk 29]
```diff
@@ -0,0 +1,24 @@
...
+    public String firstName;
+
+    @ColumnInfo(name = "last_name")
+    public String lastName;
+
+    @ColumnInfo(name = "age")
+    public int age;
+
+    @ColumnInfo(name = "weight")
+    public float weight;
```
Comment (awave1) [inline app/src/main/java/com/group15/fitnesstracker/db/User.java]:
```suggestion
    public double weight;
```

**Review by awave1 — APPROVED** [review]

**Pull request merged**